### PR TITLE
Pedal

### PR DESCRIPTION
**Pull request by joeljacobs**
Added pedal fingerprints, and _comma.dbc to Odyssey, Ridgeline, Pilot and RDX.
Altered _comma.dbc and panda/board/safety/safety_honda.h to account for variance in Pedal threshold.
Tested with 4 Pedal boards and 2 cars (RDX and Odyssey).  These threshold changes account for variance in Pedal a-to-d and possibly the physical gas pedals.

### CODE REVIEW

**Comment by joshuairl** [comment]
Yay! Thanks Joel!

File: common/fingerprints.py [hunk 1]
```diff
@@ -40,17 +42,25 @@ class GM:
...
+      57L: 3, 148L: 8, 228L: 5, 229L: 4, 316L: 8, 342L: 6, 344L: 8, 380L: 8, 399L: 7, 411L: 5, 419L: 8, 420L: 8, 427L: 3, 432L: 7, 450L: 8, 463L: 8, 464L: 8, 476L: 4, 490L: 8, 506L: 8, 542L: 7, 545L: 6, 597L: 8, 662L: 4, 773L: 7, 777L: 8, 780L: 8, 795L: 8, 800L: 8, 804L: 8, 806L: 8, 808L: 8, 817L: 4, 819L: 7, 821L: 5, 825L: 4, 829L: 5, 837L: 5, 856L: 7, 862L: 8, 871L: 8, 881L: 8, 882L: 4, 884L: 8, 891L: 8, 892L: 8, 905L: 8, 923L: 2, 927L: 8, 929L: 8, 963L: 8, 965L: 8, 966L: 8, 967L: 8, 983L: 8, 985L: 3, 1029L: 8, 1036L: 8, 1052L: 8, 1064L: 7, 1088L: 8, 1089L: 8, 1092L: 1, 1108L: 8, 1110L: 8, 1125L: 8, 1296L: 8, 1302L: 8, 1600L: 5, 1601L: 8, 1612L: 5, 1613L: 5, 1614L: 5, 1615L: 8, 1616L: 5, 1619L: 5, 1623L: 5, 1668L: 5, 513L: 6,
```
Comment (rbiasini) [inline common/fingerprints.py]: minor thing: extra spaces

File: panda/board/safety/safety_honda.h [hunk 2]
```diff
@@ -8,7 +8,7 @@
 //      brake > 0mph
 
 // these are set in the Honda safety hooks...this is the wrong place
-const int gas_interceptor_threshold = 328;
+const int gas_interceptor_threshold = 800;
```
[the remark this answers is not in the document]
Comment (rbiasini) [reply panda/board/safety/safety_honda.h]: Yeah, I get that. But this threshold should refer to the signal `INTERCEPTOR_GAS`, not `INTERCEPTOR_GAS2`. They have different factors. Makes sense?

**Comment by joeljacobs** [comment]
I wondered about that. I'll remove the panda from the PR.

File: panda/board/safety/safety_honda.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (rbiasini) [reply panda/board/safety/safety_honda.h]: [deleted my prev message]
if the dbc is at 120, with 0.2539 factor, then the threshold here should be 120/0.2539 (add 1 to be conservative). If, as you say, setting it at 600 causes sporadic cancellations, then we should figure out why :)

**Comment by joeljacobs** [comment]
Done. Added relevant pull-request for Panda.

**Comment by rbiasini** [comment]
@joeljacobs , great. Same thing for opendbc too :).

**Comment by joeljacobs** [comment]
@rbiasini Do I need to update this PR for 0.4.6? I don't think there is any problem with this PR since it's just fingerprints.

File: common/fingerprints.py [hunk 3]
```diff
@@ -40,17 +42,25 @@ class GM:
...
+    57L: 3, 145L: 8, 228L: 5, 229L: 4, 308L: 5, 316L: 8, 339L: 7, 342L: 6, 344L: 8, 380L: 8, 392L: 6, 399L: 7, 419L: 8, 420L: 8, 422L: 8, 425L: 8, 426L: 8, 427L: 3, 432L: 7, 464L: 8, 471L: 3, 476L: 4, 490L: 8, 506L: 8, 545L: 5, 546L: 3, 597L: 8, 660L: 8, 773L: 7, 777L: 8, 780L: 8, 795L: 8, 800L: 8, 804L: 8, 808L: 8, 819L: 7, 821L: 5, 829L: 5, 871L: 8, 882L: 2, 884L: 7, 892L: 8, 923L: 2, 927L: 8, 929L: 8, 963L: 8, 965L: 8, 966L: 8, 967L: 8, 983L: 8, 985L: 3, 1027L: 5, 1029L: 8, 1036L: 8, 1039L: 8, 1064L: 7, 1088L: 8, 1089L: 8, 1108L: 8, 1125L: 8, 1296L: 8, 1365L: 5, 1424L: 5, 1600L: 5, 1601L: 8, 1613L: 5, 1616L: 5, 1618L: 5, 1668L: 5, 2015L: 3, 513L: 6,
+    # sent messages
+    0xe4: 5, 0x1fa: 8, 0x200: 6, 0x30c: 8, 0x33d: 5, 0x35e: 8, 0x39f: 8, 513L: 6,
```
Comment (rbiasini) [inline common/fingerprints.py]: some of the msg adresses are repeated. Make sure that none of the "sent" messages are in the list above.

I recommend the following:
- not separating anymore between "sent messages" and the rest of the fingerprint. We only did it for the Acura ILX but it should go away (feel free to change it in this PR)
- In each fingerprint list, all the message addresses should be in decimal Long format (no hex) 
- In each fingerprint list, all the message addresses should be in monotonically increasing order (feel free to comment that 513L is the pedal)

**Comment by rbiasini** [comment]
@joeljacobs see my comment. Since some changes are needed, feel free to update to 0.4.6.

File: common/fingerprints.py (hunk 3 above)
Comment (joeljacobs) [reply common/fingerprints.py]: I noticed a lot of ids have an L after the id (835L: 4,) But a lot (especially the HEX ones) don't have the L. Do I need to add it to the hex numbers I'm converting to Dec?

**Comment by joeljacobs** [comment]
@rbiasini how does this look?

**Comment by joeljacobs** [comment]
@rbiasini This is now updated for the new refactoring. I also did what you asked and sorted all of the IDs and removed the #sent messages lines.

**Comment by joeljacobs** [comment]
I also converted all hexadecimal entries for all fingerprints, like you suggested.

**Comment by rbiasini** [comment]
is this the same as https://github.com/commaai/openpilot/pull/268? if yes, than we can close this one as I ended up verifying the correctness of PR 268. Thanks @joeljacobs

**Comment by joeljacobs** [comment]
Yes, I think he and I both changed my original PR for 0.4.7, so they should be more or less the same.

**Comment by jfrux** [comment]
Updated PR is now at #274

**Pull request closed without merging**